### PR TITLE
Adds support for running, character appearance, and EntityGUIDs

### PR DESCRIPTION
**Pull request by infiniterasadev**
Adds run_state to the character database and client packet handling for toggling running
Adds support for character appearance after creation
Sneaks in a basic handler for logout to prevent a crash
Brings over some data files from dev-creatures for packet handling later
Adds basic support for entity GUIDs

### CODE REVIEW

File: src/Rasa.Game/Data/EntityGUID.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    public struct EntityGUID : IEquatable<EntityGUID>
+    {
+        // Lower entity IDs are reserved in the client
+        const ulong _startingIdCounter = 1000;
+
+        public ulong Id { get; private set; }
+
+        public EntityGUID(EntityType entityType, ulong entityIdCounter)
+        {
+            Id = ((ulong)entityType & 0xFF) << 56 | ((_startingIdCounter + entityIdCounter) & 0x00FFFFFFFFFFFF);
```
Comment (Blumster) [inline src/Rasa.Game/Data/EntityGUID.cs]: You don't have to add the starting counter to the entity id counter, counters from 1 will be unique with the proper entity type.

File: src/Rasa.Game/Data/EntityType.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+
+namespace Rasa.Data
+{
+    public enum EntityType
+    {
+        Character   = 0,
```
Comment (Blumster) [inline src/Rasa.Game/Data/EntityType.cs]: Make sure we start the types from 1, so there are always some bits in the type in the guid.

File: src/Rasa.Game/Structures/Entity.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+﻿namespace Rasa.Structures
+{
+    using Data;
+
+    public abstract class Entity
+    {
+        public EntityGUID EntityGUID { get; set; }
+        public ulong EntityId => EntityGUID.Id;
+        public EntityType EntityType { get; set; }
```
Comment (Blumster) [inline src/Rasa.Game/Structures/Entity.cs]: Why is this here? Should this value come from the EntityGUID instance too like EntityId?

**Review by Blumster — CHANGES_REQUESTED** [review]

File: src/Rasa.Game/Structures/Entity.cs (hunk 3 above)
Comment (infiniterasadev) [reply src/Rasa.Game/Structures/Entity.cs]: I changed this, but I'm not seeing the point of encoding the type if I'm just keeping track in the property too. It looks weird. Thoughts?

File: src/Rasa.Game/Data/EntityGUID.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
+﻿using System;
+
+namespace Rasa.Data
+{
+    public struct EntityGUID : IEquatable<EntityGUID>
+    {
+        public ulong Id { get; private set; }
+        public EntityType EntityType { get; private set; }
```
Comment (Blumster) [inline src/Rasa.Game/Data/EntityGUID.cs]: Please remove this property and implement it like we've talked about in Discord. Thanks!

**Review by Blumster — CHANGES_REQUESTED** [review]

File: src/Rasa.Game/Data/EntityGUID.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+{
+    public struct EntityGUID : IEquatable<EntityGUID>
+    {
+        public ulong Raw { get; private set; }
+        public EntityType EntityType => (EntityType)(Raw >> 56);
+        public ulong Counter => Raw & 0x00FFFFFFFFFFFFFF;
+
+        public EntityGUID(EntityType entityType, ulong entityIdCounter)
+        {
+            Raw = ((ulong)entityType & 0xFF) << 56 | ((entityIdCounter) & 0x00FFFFFFFFFFFF);
```
Comment (Blumster) [inline src/Rasa.Game/Data/EntityGUID.cs]: Still a typo, please put the mask into a public const field.

**Review by Blumster — CHANGES_REQUESTED** [review]

**Review by Blumster — APPROVED** [review]
